privaterelay/views.py

```diff
@@ -138,12 +135,28 @@ def _get_event_keys_from_jwt(authentic_jwt):
     return authentic_jwt['events'].keys()
 
 
+def _handle_fxa_event(event_key, authentic_jwt, social_account):
+    if (event_key == FXA_PROFILE_CHANGE_EVENT):
+        _handle_fxa_profile_change(
+            authentic_jwt, social_account, event_key
+        )
+    if (event_key == FXA_DELETE_EVENT):
+        _handle_fxa_delete(authentic_jwt, social_account, event_key)
+
+
 def _handle_fxa_profile_change(authentic_jwt, social_account, event_key):
     token = social_account.socialtoken_set.first()
     headers = {'Authorization': 'Bearer: {0}'.format(token.token)}
     resp = requests.get(
         FirefoxAccountsOAuth2Adapter.profile_url, headers=headers
     )
+    if resp.status_code == 401:
+        resp_json = resp.json()
+        if resp_json['reason'] == 'Invalid token':
+            # TODO: use the social_account refresh token to get a new access
+            # token and retry
+            pass
```

Comment:
For this you may want to do it like this: https://github.com/pennersr/django-allauth/issues/420#issuecomment-301805706